internal/pkg/otel/manager/manager.go

```diff
@@ -361,9 +362,30 @@ func buildMergedConfig(cfgUpdate configUpdate, agentInfo info.Agent, monitoringC
 		}
 	}
 
+	if err := m.injectDiagnosticsExtension(mergedOtelCfg); err != nil {
+		return nil, fmt.Errorf("failed to inject diagnostics: %w", err)
+	}
+
 	return mergedOtelCfg, nil
 }
 
+func (m *OTelManager) injectDiagnosticsExtension(config *confmap.Conf) error {
+	extensionCfg := map[string]any{
+		"extensions": map[string]any{
+			"elastic_diagnostics": map[string]any{
+				"endpoint": paths.DiagnosticsExtensionSocket(),
+			},
+		},
+	}
+	if config.IsSet("service::extensions") {
+		extensionList := config.Get("service::extensions").([]interface{})
+		extensionList = append(extensionList, "elastic_diagnostics")
```

Comment:
It'd be safer to check if the extension is already on the list. We've had issues in the past related to extensions being added more than once.